lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -110,7 +123,8 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
           raise(ArgumentError, "with_puppet_running_on's conf_opts must be a Hash. You provided a #{conf_opts.class}: '#{conf_opts}'") if !conf_opts.kind_of?(Hash)
           cmdline_args = conf_opts[:__commandline_args__]
           service_args = conf_opts[:__service_args__] || {}
-          conf_opts = conf_opts.reject { |k,v| [:__commandline_args__, :__service_args__].include?(k) }
+          restart_when_done = conf_opts.fetch(:restart_when_done, false)
```

Comment:
I think the default behavior should be `true` since that is how puppetserver currently works.